lib/index.js

```diff
@@ -153,14 +159,32 @@ internals.initialize = function (server, next) {
 
 internals.stop = function (server, next) {
 
-    const collector = internals.state(server.root).collector;
+    const rootCollector = internals.state(server.root).collector;
 
     // Do not teardown if specifically asked not to
-    if (collector.teardownOnStop === false) {
+    if (rootCollector.teardownOnStop === false) {
         return next();
     }
 
-    return server.root.knex().destroy(next);
+    const knexConnections = [];
+
+    Object.keys(rootCollector.knexGroups).forEach((key) => {
+
+        knexConnections.push(rootCollector.knexGroups[key].knex);
+    });
+
+    // destroy() all knex instances
+    Items.parallel(knexConnections, (item, nxt) => {
+
+        item.destroy(() => {
+
+            nxt();
+        });
+    },
+    (_) => {
```

Comment:
The destroy function (same link as above -->) https://github.com/tgriesser/knex/blob/a4bfee77513d000cf0fddf130a370071c240f93d/src/client.js#L267-L286 doesn't grab the reject part of the promise there on line 268. So is it possible to catch an error from the function? I can cause an exception but I don't know how to catch and handle an error with it